openstudiocore/src/utilities/core/ApplicationPathHelpers.cxx.in

```diff
@@ -78,22 +78,34 @@ namespace openstudio {
   }
 
   openstudio::path getApplicationPath() {
-
+    openstudio::path path;
     #if defined _WIN32
       TCHAR szPath[MAX_PATH];
       if( GetModuleFileName( nullptr, szPath, MAX_PATH ) ) {
         LOG_FREE(Debug, "ApplicationPathHelpers", "getApplicationPath, szPath = '" << szPath << "'");
-        return completeAndNormalize(toPath(szPath));
+        path = toPath(szPath);
       }
     #else
       Dl_info info;
       if (dladdr("main", &info)) {
         LOG_FREE(Debug, "ApplicationPathHelpers", "getApplicationPath, info.dli_fname = '" << info.dli_fname << "'");
-        return completeAndNormalize(toPath(info.dli_fname));
+        path = toPath(info.dli_fname);
       }
     #endif
+    // TODO: JM 2019-04-18 Another option is to always start by calling findInSystemPath and modify the findInSystemPath to prepend the PATH list with '.'
+    openstudio::path applicationPath = completeAndNormalize(path);
+    if( !openstudio::filesystem::exists( applicationPath ) || openstudio::filesystem::is_directory( applicationPath ) ) {
+        LOG_FREE(Debug, "ApplicationPathHelpers", "getApplicationPath failed as is, trying in to locate in systemPath");
```

Comment:
This does not seem right to me.  We want this method to return the path of the current running application.  If path is 'OpenStudioApplication' this just assumes we are running the one in the system path?

Reply:
Seems like there should be a better way to get the full path of the current running process without having to make this assumption.

Reply:
No, you first try to location that 'OpenStudioApplication' in the current directory, then if you cannot, it means it's in the system PATH. There's no way around it IMHO. This wouldn't be a concern unless you are launching a symlink that lives in your PATH.

./OpenStudioApplicationSymlink will work
/home/OpenStudioApplicationSymlink will work
OpenStudioApplicationSymlink itself will not work if you aren't in the same folder. Cf my other comment on the PR (outside of review box)

Reply:
This is starting to make sense to me now, I'm very nervous about changes to these methods since they have caused us so much pain in the past